internal/device_managers/sfdisk/sfdisk.go

```diff
@@ -0,0 +1,249 @@
+	op.parts = append(op.parts, p)
+}
+
+func (op *Operation) DeletePartition(num int) {
+	op.deletions = append(op.deletions, num)
+}
+
+func (op *Operation) Info(num int) {
+	op.infos = append(op.infos, num)
+}
+
+// WipeTable toggles if the table is to be wiped first when committing this operation.
+func (op *Operation) WipeTable(wipe bool) {
+	op.wipe = wipe
+}
+
+// Pretend is like Commit() but uses the --no-act flag and returns the output
+func (op *Operation) Pretend() (string, error) {
+	// Handle deletions first
+	if err := op.handleDeletions(); err != nil {
+		return "", err
+	}
+
+	if err := op.handleInfo(); err != nil {
+		return "", err
+	}
+
+	script := op.buildOptions()
+	cmd := exec.Command("sh", "-c", fmt.Sprintf("echo -e \"%s\" | sudo %s --no-act %s", script, distro.SfdiskCmd(), op.dev))
+	stdout, err := cmd.StdoutPipe()
+
+	if err != nil {
+		return "", err
+	}
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return "", err
+	}
+
+	if err := cmd.Start(); err != nil {
+		return "", err
+	}
+	output, err := io.ReadAll(stdout)
+	if err != nil {
+		return "", err
+	}
+
+	errors, err := io.ReadAll(stderr)
+	if err != nil {
+		return "", err
+	}
+
+	if err := cmd.Wait(); err != nil {
+		return "", fmt.Errorf("failed to pretend to create partitions. Err: %v. Stderr: %v", err, string(errors))
+	}
+
+	return string(output), nil
+}
+
+// Commit commits an partitioning operation.
+func (op *Operation) Commit() error {
+	script := op.buildOptions()
+	if len(script) == 0 {
+		return nil
+	}
+
+	// If wipe we need to reset the partition table
+	if op.wipe {
+		// Erase the existing partition tables
+		cmd := exec.Command("sudo", distro.WipefsCmd(), "-a", op.dev)
+		if _, err := op.logger.LogCmd(cmd, "option wipe selected, and failed to execute on %q", op.dev); err != nil {
+			return fmt.Errorf("wipe partition table failed: %v", err)
+		}
+	}
+
+	op.logger.Info("running sfdisk with script: %v", script)
+	exec.Command("sh", "-c", fmt.Sprintf("echo label: gpt | sudo %s %s", distro.SfdiskCmd(), op.dev))
+	cmd := exec.Command("sh", "-c", fmt.Sprintf("echo \"%s\" | sudo %s %s", script, distro.SfdiskCmd(), op.dev))
+	if _, err := op.logger.LogCmd(cmd, "deleting %d partitions and creating %d partitions on %q", len(op.deletions), len(op.parts), op.dev); err != nil {
+		return fmt.Errorf("create partitions failed: %v", err)
+	}
+
+	return nil
+}
+
+// ParseOutput takes the output from sfdisk. Similarly to sgdisk
+// it then uses regex to parse the output into understood values like 'start' 'size' and attempts
+// to catch any failures and wrap them to return to the caller.
+func (op *Operation) ParseOutput(sfdiskOutput string, partitionNumbers []int) (map[int]device_managers.Output, error) {
+	if len(partitionNumbers) == 0 {
+		return nil, nil
+	}
+
+	// Prepare the data, and a regex for matching on partitions
+	partitionRegex := regexp.MustCompile(`^/dev/\S+\s+\S*\s+(\d+)\s+(\d+)\s+\d+\s+\S+\s+\S+\s+\S+.*$`)
```

Comment:
This deserves some explanation.

Also, I feel like we should ignore anything before the "New situation:" line.

Reply:
Added some details